app/models/product_importer.rb

```diff
@@ -177,7 +181,12 @@ def count_existing_products
           AND spree_variants.deleted_at IS NULL', supplier_id).
           count
 
-        @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        if @supplier_products[:by_supplier][supplier_id]
+          @supplier_products[:by_supplier][supplier_id][:existing_products] = products_count
+        else
+          @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        end
```

Comment:
Is there an easy way to create blank/generic objects and assign attributes? Or do you mean creating a whole new class for storing this data? Really it's just a hash containing a few values that are all integers. I could possibly restructure the hash to make it simpler.

One thing I've done this week is write a lot more specs that test the functionality in a wide range of different scenarios. That makes it easier to tell if something isn't working after a change.